npm run watch fails with unresolved dependencies

Ubuntu 14.04

- Follow the instructions in https://github.com/redhat-developer/vscode-java/blob/master/CONTRIBUTING.md#building-eclipse--jdt-language-server
`npm run build-server` works but `npm run watch-server` fails with

Cannot resolve dependencies of product languageServer.product:
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.jdt.launching.macosx" and version matching "0.0.0" was not found in the target platform
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.core.net.linux.x86" and version matching "0.0.0" was not found in the target platform
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.core.net.win32.x86" and version matching "0.0.0" was not found in the target platform
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.core.net.win32.x86_64" and version matching "0.0.0" was not found in the target platform
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.equinox.security.macosx" and version matching "0.0.0" was not found in the target platform
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.equinox.security.win32.x86" and version matching "0.0.0" was not found in the target platform
[ERROR]   eclipse-plugin artifact with ID "org.eclipse.equinox.security.win32.x86_64" and version matching "0.0.0" was not found in the target platform

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -3,6 +3,7 @@ const gulp = require('gulp');
 const cp = require('child_process');
 const decompress = require('gulp-decompress');
 const download = require('gulp-download');
+const glob = require('glob');
 const fse = require('fs-extra');
 const path = require('path');
 const server_dir = '../eclipse.jdt.ls';
@@ -27,26 +28,22 @@ gulp.task('build_server', function(done) {
 });
 
 gulp.task('dev_server', function(done) {
-	let command = mvnw() +' -Pserver-distro,fast -o clean package ';
-	if(isLinux()){
-		command +='-Denvironment.os=linux -Denvironment.ws=gtk -Denvironment.arch=x86_64';
-	}
-	else if(isMac()){
-		command += '-Denvironment.os=macosx -Denvironment.ws=cocoa -Denvironment.arch=x86_64';
-	}
-	else if(isWin()){
-		command += '-Denvironment.os=win32 -Denvironment.ws=win32 -Denvironment.arch=x86_64';
-	}
+	let command = mvnw() +' -o -pl org.eclipse.jdt.ls.core,org.eclipse.jdt.ls.target clean package';
 	console.log('executing '+command);
 	cp.execSync(command, {cwd:server_dir, stdio:[0,1,2]} );
-	gulp.src(server_dir +'/org.eclipse.jdt.ls.product/distro/*.tar.gz')
-		.pipe(decompress())
-		.pipe(gulp.dest('./server'))
+	glob.Glob(server_dir +'/org.eclipse.jdt.ls.core/target/org.eclipse.jdt.ls.core-*-SNAPSHOT.jar',
+	    (_error, sources) => {
+			glob.Glob('./server/plugins/org.eclipse.jdt.ls.core_*.jar',
+				(_error, targets) => {
+					console.log('Copying '+sources[0]+ ' to '+targets[0]);
+					fse.copy(sources[0], targets[0]);
+			});
+	});
 	done();
 });
 
 gulp.task('watch_server',function(done) {
-	gulp.watch(server_dir+'/org.eclipse.jdt.ls.core/**/*.java',['dev_server']);
+	gulp.watch(server_dir+'/org.eclipse.jdt.ls.core/**/*.java', gulp.series('dev_server'));
 	done();
 });
 

